### PR TITLE
make staff column disable on esg enable and show demo on esg enable

### PR DESCRIPTION
**Pull request by leangseu-edx**
**TL;DR -** Make staff column disable on esg enable and show demo on esg enable

JIRA:
- [AU-593](https://openedx.atlassian.net/browse/AU-593)
- [AU-570](https://openedx.atlassian.net/browse/AU-570)

**What changed?**

- ESG flag enabled

Screen Shot 2022-04-06 at 3 21 26 PM

- ESG flag disabled

Screen Shot 2022-04-06 at 3 22 39 PM

**Developer Checklist**

- [ ] Reviewed the [release process](https://github.com/edx/edx-ora2/blob/master/.github/release_process.md)
- [ ] Translations and JS/SASS compiled
- [ ] Bumped version number in [setup.py](https://github.com/edx/edx-ora2/blob/a62e81a9b0d89223476967ec3c27f3557a850735/setup.py#L39) and [package.json](https://github.com/edx/edx-ora2/blob/a62e81a9b0d89223476967ec3c27f3557a850735/package.json#L3)

**Testing Instructions**

- Enabled ESG
  - Go to [http://localhost:18000/admin/waffle/flag/](http://localhost:18000/admin/waffle/flag/)
  - add/update Flag
    - name `openresponseassessment.enhanced_staff_grader`
    - change everyone to `yes`
    - check `staff`
    - Save
  - [ ] Go to `instructor dashboard -> Open Responses`
  - [ ] staff column should be disabled
  - [ ] `staff_grader` column should show esg link

- Disabled ESG
  - Go to [http://localhost:18000/admin/waffle/flag/](http://localhost:18000/admin/waffle/flag/)
  - add/update Flag
    - name `openresponseassessment.enhanced_staff_grader`
    - change everyone to `no`
    - check `staff`
    - Save
  - [ ] Go to `instructor dashboard -> Open Responses`
  - [ ] staff column should behave like it used to
  - [ ] `staff_grader` column should show esg link to demo

**Reviewer Checklist**

Collectively, these should be completed by reviewers of this PR:

- [ ] I've done a visual code review
- [ ] I've tested the new functionality

FYI: @openedx/content-aurora

### CODE REVIEW

**Review by jansenk — COMMENTED** [review]
two small nits but overall looks good! I want to manually test before giving an actual approval

**Review by mattcarter — APPROVED** [review]
approved

File: openassessment/xblock/static/js/src/lms/oa_course_items_listing.js [hunk 1]
```diff
@@ -44,6 +46,28 @@ export class CourseItemsListingView {
...
 
+    const esgEnabled = context.ENHANCED_STAFF_GRADER;
+    const esgRootUrl = context.ORA_GRADING_MICROFRONTEND_URL;
+
+    const ESGCell = Backgrid.UriCell.extend({
+      render() {
+        this.$el.empty();
+        const name = this.column.get('name');
+        this.$el.addClass(name);
+        const displayValue = itemViewEnabled && esgEnabled ? gettext('View and grade responses') : gettext('Demo the new Grading Experience');
```
Comment (jansenk) [inline openassessment/xblock/static/js/src/lms/oa_course_items_listing.js:57]: I think this is wrong. I don't think we should care about itemViewEnabled here.

if (!itemViewEnabled, esgEnabled), this would show "Demo the new Grading Experience" which is not correct.

**Review by jansenk — CHANGES_REQUESTED** [review]
Just one change

File: openassessment/xblock/static/js/src/lms/oa_course_items_listing.js (hunk 1 above)
Comment (jansenk) [reply openassessment/xblock/static/js/src/lms/oa_course_items_listing.js:57]: itemViewEnabled is a feature where we render content from the Xblock into the page, which is what we're doing in the cases where we were doing the in-page staff and waiting step stuff. This is just an external link

**Review by jansenk — APPROVED** [review]
